osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -94,6 +97,35 @@ static inline void flagsFromSysfs(const std::string& name, size_t& flags) {
     }
   }
 }
+#else //  Apple || FreeBSD
+// Based on IFM_SUBTYPE_ETHERNET_DESCRIPTIONS in if_media.h
+static int get_linkspeed(int ifm_subtype) {
+  switch(ifm_subtype)  {
+    case IFM_10_T : return 10;
+    case IFM_10_2 : return 10;
+    case IFM_10_5 : return 10;
+    case IFM_100_TX : return 100;
+    case IFM_100_FX : return 100;
+    case IFM_100_T4 : return 100;
+    case IFM_100_VG : return 100;
+    case IFM_100_T2 : return 100;
+    case IFM_1000_SX : return 1'000;
+    case IFM_10_STP : return 10;
+    case IFM_10_FL : return 10;
+    case IFM_1000_LX : return 1'000;
+    case IFM_1000_CX : return 1'000;
+    case IFM_1000_T : return 1'000;
+    case IFM_HPNA_1 : return 1;
+    case IFM_10G_SR : return 10'000;
+    case IFM_10G_LR : return 10'000;
+    case IFM_10G_CX4 : return 10'000;
+    case IFM_10G_T : return 10'000;
+    case IFM_2500_T : return 2'500;
+    case IFM_5000_T : return 5'000;
```

Comment:
For something this simple, falling through to the next case is probably easier to read:

```
  case IFM_HPNA_1:
    return 1;
  case IFM_10_T:
  case IFM_10_2:
  case IFM_10_5:
  case IFM_10_STP:
  case IFM_10_FL:
    return 10;
  case IFM_100_TX:
  case IFM_100_FX:
  case IFM_100_T4:
  case IFM_100_VG:
  case IFM_100_T2:
  case IFM_1000_SX:
  case IFM_1000_LX:
  case IFM_1000_CX:
  case IFM_1000_T:
    return 1'000;
  case IFM_2500_T:
    return 2'500;
  case IFM_5000_T:
    return 5'000;
  case IFM_10G_SR:
  case IFM_10G_LR:
  case IFM_10G_CX4:
  case IFM_10G_T:
    return 10'000;
```